src/main/java/dev/j3fftw/litexpansion/machine/Generator.java

```diff
@@ -0,0 +1,143 @@
+
+import dev.j3fftw.litexpansion.Items;
+import io.github.thebusybiscuit.slimefun4.implementation.items.electric.generators.CoalGenerator;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.MachineFuel;
+import org.bukkit.Material;
+import org.bukkit.Tag;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.Nonnull;
+
+public class Generator extends CoalGenerator {
+
+    public Generator() {
+        super(Items.LITEXPANSION, Items.GENERATOR, RecipeType.ENHANCED_CRAFTING_TABLE, new ItemStack[] {
+            null, Items.RE_BATTERY, null,
+            null, Items.MACHINE_BLOCK, null,
+            null, new ItemStack(Material.FURNACE), null
+        });
+    }
+
+    @Nonnull
+    @Override
+    public String getInventoryTitle() {
+        return "&7Generator";
+    }
+
+    @Nonnull
+    @Override
+    public ItemStack getProgressBar() {
+        return new ItemStack(Material.FIRE_CHARGE);
+    }
+
+    @Override
+    public int getEnergyProduction() {
+        return 10;
+    }
+
+    @Override
+    protected void registerDefaultFuelTypes() {
+
+        registerFuel(new MachineFuel(100, new ItemStack(Material.LAVA_BUCKET)));
+        registerFuel(new MachineFuel(80, new ItemStack(Material.COAL_BLOCK)));
+        registerFuel(new MachineFuel(20, new ItemStack(Material.DRIED_KELP_BLOCK)));
+        registerFuel(new MachineFuel(12, new ItemStack(Material.BLAZE_ROD)));
+        registerFuel(new MachineFuel(8, new ItemStack(Material.COAL)));
+        registerFuel(new MachineFuel(8, new ItemStack(Material.CHARCOAL)));
+
+        for (Material mat : Tag.ITEMS_BOATS.getValues()) {
+            registerFuel(new MachineFuel(6, new ItemStack(mat)));
+        }
+
+        registerFuel(new MachineFuel(6, new ItemStack(Material.SCAFFOLDING)));
+
+        for (Material mat : Tag.LOGS.getValues()) {
+            registerFuel(new MachineFuel(2, new ItemStack(mat)));
+        }
+
+        for (Material mat : Tag.PLANKS.getValues()) {
+            registerFuel(new MachineFuel(1, new ItemStack(mat)));
+        }
+
+        for (Material mat : Tag.WOODEN_SLABS.getValues()) {
+            registerFuel(new MachineFuel(1, new ItemStack(mat)));
+        }
+
+        for (Material mat : Tag.WOODEN_BUTTONS.getValues()) {
+            registerFuel(new MachineFuel(1, new ItemStack(mat)));
+        }
+
+        for (Material mat : Tag.WOODEN_FENCES.getValues()) {
+            registerFuel(new MachineFuel(1, new ItemStack(mat)));
+        }
+
+        for (Material mat : Tag.WOODEN_TRAPDOORS.getValues()) {
+            registerFuel(new MachineFuel(1, new ItemStack(mat)));
+        }
+
+        for (Material mat : Tag.WOODEN_PRESSURE_PLATES.getValues()) {
+            registerFuel(new MachineFuel(1, new ItemStack(mat)));
+        }
+        for (Material mat : Tag.WOODEN_DOORS.getValues()) {
+            registerFuel(new MachineFuel(1, new ItemStack(mat)));
+        }
```

Comment:
Most of these things should really just be added to the base Coal Generator to be honest. There is no need to have an extra Generator for something the existing one basically already does or should do.

Reply:
select what you want added and i will do that but i would need the generator in here anyways since some recipes require it

Reply:
Ah okay, yeah feel free to make a PR for all the wooden Tags, I'd only add those in my opinion.
But you may also have to add the tags to the unit test fake ones too

Reply:
eeee i will add the wooden tags and you do unit test deal?

Reply:
no